src/bdiff.c

```diff
@@ -69,6 +82,39 @@ static unsigned find_end_delta(
     return end_delta;
 }
 
+static unsigned slidey_aligner(
+        unsigned const sample_size, data_seeker const ds,
+        data_fetcher const df, char * const buf_fixed,
+        char * const buf_sliding, void * const fixed, void * const sliding,
+        unsigned const fixed_start, unsigned const sliding_end,
+        unsigned slide_distance) {
+    for (; slide_distance; slide_distance--) {
+        ds(sliding, sliding_end - slide_distance);
+        ds(fixed, fixed_start);
+        assert(df(fixed, buf_fixed, 1) == 1);
+        assert(df(sliding, buf_sliding, 1) == 1);
+        unsigned i = 0;
```

Comment:
I presume this is defined here because in C the loop variable is otherwise only available in the scope of the loop?